src/sentry/data_export/base.py

```
@@ -27,29 +27,29 @@ class ExportQueryType:
     DISCOVER_STR = "Discover"
 
     @classmethod
-    def as_choices(cls):
+    def as_choices(cls) -> tuple[tuple[int, str], tuple[int, str]]:
         return (
             (cls.ISSUES_BY_TAG, str(cls.ISSUES_BY_TAG_STR)),
             (cls.DISCOVER, str(cls.DISCOVER_STR)),
         )
 
     @classmethod
-    def as_str_choices(cls):
+    def as_str_choices(cls) -> tuple[tuple[str, str], tuple[str, str]]:
         return (
             (cls.ISSUES_BY_TAG_STR, cls.ISSUES_BY_TAG_STR),
             (cls.DISCOVER_STR, cls.DISCOVER_STR),
         )
 
     @classmethod
-    def as_str(cls, integer):
+    def as_str(cls, integer: int) -> str:
         if integer == cls.ISSUES_BY_TAG:
             return cls.ISSUES_BY_TAG_STR
-        elif integer == cls.DISCOVER:
+        else:
             return cls.DISCOVER_STR
 
     @classmethod
-    def from_str(cls, string):
+    def from_str(cls, string: str) -> int:
         if string == cls.ISSUES_BY_TAG_STR:
             return cls.ISSUES_BY_TAG
-        elif string == cls.DISCOVER_STR:
+        else:
             return cls.DISCOVER
```

there aren't any other choices in this enum but would it be better to raise an error if the arg is invalid instead of defaulting to the results for discover?

---

Yes, I think it would be better to `assert` and have it fail loudly. Even if everything works correctly today, I feel like having it default to discover could become a hard-to-track bug a year from now.